lib/packages/docutils/highlite.nim

```diff
@@ -190,31 +190,33 @@ proc nimNextToken(g: var GeneralTokenizer, keywords: openArray[string] = @[]) =
   var pos = g.pos
   g.start = g.pos
   if g.state == gtStringLit:
-    g.kind = gtStringLit
-    while true:
+    if g.buf[pos] == '\\':
+      g.kind = gtEscapeSequence
+      inc(pos)
       case g.buf[pos]
-      of '\\':
-        g.kind = gtEscapeSequence
+      of 'x', 'X':
```

Comment:
* pre-existing (maybe?) but what about other escape sequences, eg:
```nim
proc main*()=
  ## \x123
  ## `\x123`
  runnableExamples:
    echo "abc\x1234"
    echo "abc\x1"
    echo "abc\uabcd3" # bug with this one
    echo "abc\u{12345}def" # and this one
    echo "abc\e"
main()
```

is there possibility to reuse code (possibly via a lib/std/private/lexerutils) to avoid duplication with what compiler already does?

Reply:
I just enabled the existing code by moving case of first `\` into surrounding `if` — Github obviously shows more changes than has been really done.

I can hardly guess why `highlite.nim` was written with its own separate lexer.

@Araq ?

Reply:
Lexer: Transforms  "\n"  into  \10. Skips comments and whitespace (more or less).
Highlighter: Highlights "\n". Highlights comments and whitespace.

Different code for different things.